classes/jplain.lua

```diff
@@ -17,13 +17,25 @@ function jplain:_j_common ()
   self:loadPackage("font-fallback")
   self:loadPackage("hanmenkyoshi")
   self:registerPostinit(function (class)
-      class:bidiDisableTypesetter(SILE.typesetter)
-      class:bidiDisableTypesetter(SILE.defaultTypesetter)
-      SILE.call("font:add-fallback", { family = "Noto Sans CJK JP" })
-    end)
+    class:bidiDisableTypesetter(SILE.typesetter)
+    class:bidiDisableTypesetter(SILE.defaultTypesetter)
+  end)
   self.defaultFrameset.content.tate = self.options.layout == "tate"
   self:declareHanmenFrame("content", self.defaultFrameset.content)
   SILE.settings:set("document.parindent", SILE.nodefactory.glue("10pt"))
+  if SILE.settings:get("document.language") ~= "ja" then
+    SU.deprecated("document.language ≠ \"ja\" & jplain:…", nil, "0.13.2", "0.15.0", [[
+  Prior to SILE v0.13.2, `jplain`, despite its name, enforced the
+  language being set te Japanese. It no longer makes this assumption.
+  To use a class like `jplain` for other languages, please base a
+  *new* custom class inheriting from `jplain` rather than `jplain`
+  directly. To use `jplain` for Japanese, you *must* specify
+  \\language[main=ja]. (Also you'll probably want to set a font!)
+    ]])
+    SILE.languageSupport.loadLanguage("ja")
+    SILE.settings:set("document.language", "ja", true)
+    SILE.settings:set("font.family", "Noto Sans CJK JP", true)
```

Comment:
I refactored this just a bit to set the default font instead of the fallback font to support Japaneese. I realize this is just a shim about to go away, but this seems like a better example for somebody that wanted to know how to do this in their own class later. Does this make sense or am I missing something and leaving Gentium as the default font while the language is set to `ja` actually makes sense?

Reply:
You're not missing anything.